frame/treasury/src/lib.rs

```diff
@@ -625,6 +775,324 @@ decl_module! {
+		/// # </weight>
+		#[weight = T::WeightInfo::assign_curator()]
+		fn assign_curator(
+			origin,
+			#[compact] bounty_id: ProposalIndex,
+			curator: <T::Lookup as StaticLookup>::Source,
+			#[compact] fee: BalanceOf<T>,
+		) {
+			T::ApproveOrigin::ensure_origin(origin)?;
+
+			let curator = T::Lookup::lookup(curator)?;
+			Bounties::<T>::try_mutate_exists(bounty_id, |maybe_bounty| -> DispatchResult {
+				let mut bounty = maybe_bounty.as_mut().ok_or(Error::<T>::InvalidIndex)?;
+				match bounty.status {
+					BountyStatus::Funded | BountyStatus::CuratorAssigned { .. } => {},
+					_ => return Err(Error::<T>::UnexpectedStatus.into()),
+				};
+
+				bounty.status = BountyStatus::CuratorAssigned { curator };
+				bounty.fee = fee;
+
+				Ok(())
+			})?;
+		}
+
+		/// Unassign curator from a bounty.
+		/// If the bounty is on active or pending payout status, the curator deposit will be slashed from curator.
+		///
+		/// May only be called from `T::ApproveOrigin` or the curator.
+		///
+		/// # <weight>
+		/// - O(1).
+		/// - Limited storage reads.
+		/// - One DB change.
+		/// # </weight>
+		#[weight = T::WeightInfo::unassign_curator()]
+		fn unassign_curator(
+			origin,
+			#[compact] bounty_id: ProposalIndex,
+		) {
+			let maybe_curator = ensure_signed(origin.clone())
+				.map(Some)
+				.or_else(|_| T::ApproveOrigin::ensure_origin(origin).map(|_| None))?;
+
+			Bounties::<T>::try_mutate_exists(bounty_id, |maybe_bounty| -> DispatchResult {
+				let mut bounty = maybe_bounty.as_mut().ok_or(Error::<T>::InvalidIndex)?;
+				match bounty.status {
+					BountyStatus::CuratorAssigned { ref curator } => {
+						ensure!(maybe_curator.map_or(true, |c| c == *curator), BadOrigin);
+					},
+					BountyStatus::Active { ref curator, .. } | BountyStatus::PendingPayout { ref curator, .. } => {
+						ensure!(maybe_curator.map_or(true, |c| c == *curator), BadOrigin);
+						let _ = T::Currency::slash_reserved(curator, bounty.curator_deposit);
+						bounty.curator_deposit = 0.into();
+					},
+					_ => return Err(Error::<T>::UnexpectedStatus.into()),
+				};
+
+				bounty.status = BountyStatus::Funded;
+				Ok(())
+			})?;
+		}
+
+		/// Accept the curator role for a bounty.
+		/// A deposit will be reserved from curator and refund upon successful payout.
+		///
+		/// May only be called from the curator.
+		///
+		/// # <weight>
+		/// - O(1).
+		/// - Limited storage reads.
+		/// - One DB change.
+		/// # </weight>
+		#[weight = T::WeightInfo::accept_curator()]
+		fn accept_curator(origin, #[compact] bounty_id: ProposalIndex) {
+			let signer = ensure_signed(origin)?;
+
+			Bounties::<T>::try_mutate_exists(bounty_id, |maybe_bounty| -> DispatchResult {
+				let mut bounty = maybe_bounty.as_mut().ok_or(Error::<T>::InvalidIndex)?;
+
+				match bounty.status {
+					BountyStatus::CuratorAssigned { ref curator } => {
+						ensure!(signer == *curator, Error::<T>::RequireCurator);
+
+						let deposit = T::BountyCuratorDeposit::get() * bounty.fee;
```

Comment:
I see nowhere in the code where this deposit is unreserved. something missing no ?

Reply:
you are right. I missed it.